core/src/main/java/io/questdb/std/str/Utf16Sink.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std.str;
+
+import io.questdb.std.Unsafe;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static io.questdb.std.Numbers.hexDigits;
+
+public interface Utf16Sink extends CharSink<Utf16Sink> {
+
+    /**
+     * Special behaviour of UTF16 sinks is to tread byte as INT
```

Comment:
Worth having a comment around this special behavior in the `CharSink` interface's method declaration

Reply:
removed `put(byte)` from Utf16Sink, this was asking for trouble. Done in https://github.com/questdb/questdb/pull/4137/commits/4f372f37e5cf4d5b7f9e8374383a72b79d8eee10